src/transformers/models/phi/modeling_phi.py

```diff
@@ -242,18 +258,21 @@ def __init__(self, config: PhiConfig, layer_idx: Optional[int] = None):
                 f"hidden_size must be divisible by num_heads (got `hidden_size`: {self.hidden_size}"
                 f" and `num_heads`: {self.num_heads})."
             )
-        self.query_key_value = nn.Linear(self.hidden_size, 3 * self.hidden_size, bias=True)
+
+        self.q_proj = nn.Linear(self.hidden_size, self.num_heads * self.head_dim, bias=True)
+        self.k_proj = nn.Linear(self.hidden_size, self.num_key_value_heads * self.head_dim, bias=True)
+        self.v_proj = nn.Linear(self.hidden_size, self.num_key_value_heads * self.head_dim, bias=True)
         self.dense = nn.Linear(self.num_heads * self.head_dim, self.hidden_size, bias=True)
-        self.qk_layernorm = config.qk_layernorm
 
+        self.qk_layernorm = config.qk_layernorm
```

Comment:
I don't remember if this is used for PHI or if it was added to have copied from, we probably can't remove it now anyway!

Reply:
It wasn't used for Phi, but it was added by susnato on the current Phi code. It could be worth keeping it, since we might use it later on.

Reply:
Alright 👍🏻